### PR TITLE
fix: don't retry a request that is aborted intentionally

### LINKED ISSUE #120: Cancellation causes retry with default retry handler
#### Steps to reproduce

  1. Configure gaxios with an onRetryAttempt handler that logs to console.
  2. Create a gaxios request and then use the cancellation api (AbortController) to cancel it.
  3. Note that the console output indicates the request is an AbortError, and gaxios will attempt to retry the request.

I think this is likely an omission in `shouldRetryRequest`. While this function can be overridden, the default behavior is probably surprising.

### PR DESCRIPTION
**Pull request by bcoe**
If a user has explicitly aborted a request, let's not retry it.

fixes #120

### CODE REVIEW

File: src/common.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ import {URL} from 'url';
 
 export class GaxiosError<T = any> extends Error {
   code?: string;
+  type?: string;
```
Comment (JustinBeckwith) [inline src/common.ts]: Could we add a header here to come some idea of what the expected contents would be?

File: src/retry.ts [hunk 2]
```diff
@@ -89,6 +89,12 @@ export async function getRetryConfig(err: GaxiosError) {
 function shouldRetryRequest(err: GaxiosError) {
   const config = getConfig(err);
 
+  // If the user has explicitly aborted the request, don't retry.
+  // "type" is populated on the error objects created by node-fetch.
+  if (err.type === 'aborted') {
```
Comment (JustinBeckwith) [inline src/retry.ts]: Makes me wonder - should we type the `type` above as `type?: 'ABORTED';`?

Comment (stephenplusplus) [inline src/retry.ts]: So node-fetch is putting the “aborted” on... I guess a small concern would be that they change that at some point. Is it an advertised behavior that you think their versioning would guarantee?

Comment (bcoe) [reply src/retry.ts]: in their docs, it looks like they advocate using err.name, I've updated accordingly:

https://github.com/bitinn/node-fetch#request-cancellation-with-abortsignal

File: src/common.ts (hunk 1 above)
Comment (bcoe) [reply src/common.ts]: reading their docs, it looks like they actually advocate using `err.name`:

https://github.com/bitinn/node-fetch#request-cancellation-with-abortsignal

File: src/retry.ts (hunk 2 above)
Comment (bcoe) [reply src/retry.ts]: opted for an alternate approach in which we use `err.name`.

**Comment by bcoe** [comment]
@acchou is this the fix you had in mind?

**Comment by acchou** [comment]
Yes, this works.

**Pull request merged**